ngraph_bridge/ngraph_rewrite_pass.h

```diff
@@ -13,14 +13,34 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  *******************************************************************************/
+
 #pragma once
 
-#include "tensorflow/core/graph/graph.h"
+#include "tensorflow/core/common_runtime/optimization_registry.h"
 
 namespace tensorflow {
 namespace ngraph_bridge {
 
-Status AddIdentityN(Graph* graph, std::set<string> skip_these_nodes);
+class NGraphRewritePass : public GraphOptimizationPass {
+ public:
+  NGraphRewritePass() = default;
+  ~NGraphRewritePass() override = default;
+
+  Status Rewrite(Graph* graph, std::set<string> skip_these_nodes = {},
```

Comment:
Is the separate Rewrite method needed to share code with optimizer? Is it not possible to use the Run method?

Reply:
Yes, and no. Please take a look at the definition of the two functions.